src/adaptor/ini.rs

```diff
@@ -159,14 +141,36 @@ fn insert_or_expand(map: &mut Map<String, Value>, key: String, value: Value) {
     }
 }
 
+/// Inserts all the values of a section into the configuration heirarchy
+/// Performs recursive lookups on the `Map` until the correct object is found
+fn insert_section(root: &mut Map<String, Value>, section_name: &str, pairs: Vec<(&str, &str)>) {
+    // recursively query the object using the split section name
+    let mut insert_map = section_name
+        .split(".")
```

Comment:
What will happen with something like `[foo    .bar   .  baz]`? Can we trim strings here?